python_modules/dagster/dagster_tests/asset_defs_tests/test_assets_job.py

```diff
@@ -1941,3 +1942,47 @@ def unpartitioned_asset():
         frozenset(["hourly_asset", "unpartitioned_asset"]),
         frozenset(["daily_asset_different_start_date", "unpartitioned_asset"]),
     }
+
+
+@asset(partitions_def=StaticPartitionsDefinition(["a", "b", "c"]))
+def static_partition_asset():
+    return 0
+
+
+partitioned_asset_job = build_assets_job("static_partition_asset_job", [static_partition_asset])
+
+
+def reconstruct_partitioned_asset_job():
+    return partitioned_asset_job
+
+
+def test_asset_materialization_planned():
```

Comment:
I think `test_asset_events` might be a better place for this test. Also, I think we might e able to avoid building a job and instead just call `materialize_to_memory(instance=instance...)`.